### ISSUE TITLE
[Feature Request] Default game controller mappings for Hardkernel's ODROID series of handheld gaming consoles (ODROID-Go Ultra, ODROID-Go Super, and ODROID-GO Advance)

### ISSUE THREAD

**Issue by Hedda** [kodi-game/controller-topology-project#173]
Requesting default game controller mappings for ODROID-Go Ultra, ODROID-Go Super, and ODROID-GO Advance from Hardkernel.

"OGU", "OGS" and "OGA" are Hardkernel's three popular ARM-based handheld gaming consoles running Linux operating systems.

- ODROID-Go Ultra (OGU) is the most powerful at $111(US) -> https://www.hardkernel.com/shop/odroid-go-ultra-dim-gray/ 
  - (OGU clear/white color variant -> https://www.hardkernel.com/shop/odroid-go-ultra-clear-white/)
    * ODROID official forum discussion on OGU with detailed specifications -> https://forum.odroid.com/viewtopic.php?t=45143
- ODROID-Go Super (OGS) is the mid-range at $80(US) -> https://www.hardkernel.com/shop/odroid-go-super-dim-gray/
   - (OGS clear/white color variant -> https://www.hardkernel.com/shop/odroid-go-super-clear-white/)
    * ODROID official forum discussion on OGU with detailed specifications -> https://forum.odroid.com/viewtopic.php?f=187&t=41283
- ODROID-Go Advance (OGA) Revision 1.1 is the budget version requiring DIY-assembly $59(US) -> https://www.hardkernel.com/shop/odroid-go-advance-black-edition-clear-white/

![image](https://user-images.githubusercontent.com/6320001/195067611-3535250b-1d22-4871-ac8d-b2db23781c70.png)

OGU, OGS, feature the same inputs while OGA Rev 1.1 is missing Analog joystick2, and OGA Rev 1.0 is also missing shoulder buttons:

- ODROID-Go Ultra (OGU) Input Buttons = F1, F2, F3, F4, F5, F6, A, B, X, Y, Direction Pad, Left Shoulder, Right Shoulder, Left Shoulder2, Right Shoulder2, Analog joystick, Analog joystick2
- ODROID-Go Super (OGS) Input Buttons = F1, F2, F3, F4, F5, F6, A, B, X, Y, Direction Pad, Left Shoulder, Right Shoulder, Left Shoulder2, Right Shoulder2, Analog joystick, Analog joystick2
- ODROID-Go Advance (OGA) Revision 1.1 Input Buttons = F1, F2, F3, F4, F5, F6, A, B, X, Y, Direction Pad, Left Shoulder, Right Shoulder, Left Shoulder2, Right Shoulder2, Analog joystick
- ODROID-Go Advance (OGA) Revision 1.0 Input Buttons = F1, F2, F3, F4, F5, F6, A, B, X, Y, Direction Pad, Left Shoulder, Right Shoulder, Analog joystick

Back story: Kodi and RetroPlayer already work on these portable game consoles but there are no default game controller mappings for their input (buttons and joystick). Their hardware comes preinstalled with standard Linux (Ubuntu) however they also supports running [Lakka ](https://www.lakka.tv/) or [EmuELEC](https://github.com/EmuELEC/EmuELEC) (which are purposely built "just enough operating system" Linux distros for retro emulation gaming based on [LibreELEC](https://libreelec.tv/) and [CoreELEC](https://coreelec.org) but with [RetroArch](https://www.retroarch.com/) as [EmulationStation](https://emulationstation.org/) as front-ends respectively, though both feature options to install Kodi with RetroPlayer).

https://wiki.odroid.com/odroid_go_ultra/odroid_go_ultra

https://wiki.odroid.com/odroid_go_super/

https://wiki.odroid.com/odroid_go_advance/

@garbear FYI, [odroid](https://forum.odroid.com/memberlist.php?mode=viewprofile&u=55) who is the forum site admin who works for Hardkernel have previously said that they are willing to send you and Team Kodi hardware samples and if you are interested then you only contact them with a "brief porting plan" for default input mapping.

https://forum.odroid.com/viewtopic.php?f=187&t=42710&p=329999

PS: Check out this Kodi forum thread for other similar ARM-based handheld gaming consoles running Linux operating systems:

https://forum.kodi.tv/showthread.php?tid=357351

**Comment by garbear** [comment]
I'm happy to accept an ODROID in exchange for full input support. I prefer the slowest models to flush out any performance issues. I also asked around the team to see if anyone else is interested.

**Comment by Hedda** [comment]
> I'm happy to accept an ODROID in exchange for full input support. I prefer the slowest models to flush out any performance issues. I also asked around the team to see if anyone else is interested.

@garbear  That sounds awesome! Can you please see if you can send odroid site admin (hardkernel employee) your contact and shipping information via private message or e-mail? Check: 

https://forum.odroid.com/viewtopic.php?f=187&t=42710&p=356909

and

https://forum.odroid.com/memberlist.php?mode=viewprofile&u=55

**Comment by Hedda** [comment]
Oh, regarding performance; both ODROID-Go Advance (OGA) and the ODROID-Go Super (OGS) use same RockChip RK3326 SoC, and other than build quality the main difference between them is a much higher resolution display in OGS + a second joystick.

Only the new ODROID-Go Ultra (OGU) has been upgraded to use Amlogic S922X instead.

@garbear  Note that not sure if ODROID-Go Advance (OGA) can be used with Kodi because it has a very low-resolution display, so practically I would suggest really only recommending using Kodi with ODROID-Go Super (OGS) and/or ODROID-Go Ultra (OGU).

**Comment by Hedda** [comment]
@garbear Did you make contact with someone at Hardkernel about ODROID-Go samples donation for Kodi controller mappings?

**Comment by garbear** [comment]
No, work picked up and I don't have much time for Kodi beyond code review. I can't say when things will die down, but if they do I'll report back.
